palindrome_linked_list.cpp

```diff
@@ -0,0 +1,152 @@
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+#include <iostream>
+#include <stack>
+#include <string>
+
+using namespace std;
+
+// basic linked list class
+class linkedList {
+public:
+
+	class node {
+	public:
+		char val;
+		node * next;
+		
+		node ( char c ) {
+			val = c;
+			next = NULL;
+		}
+
+	};
+	
+	node * head;
+
+	linkedList() {
+		head = NULL;
+	}
+
+	node * add( node * prev, char c ) {
+		
+		node * new_node = new node( c );
+		
+		if ( prev != NULL ) {
+			prev->next = new_node;
+		}
+		else {
+			head = new_node;
+		}
+		
+		return new_node;
+	}
+	
+	// make linked list out of string
+	make_list( string input ) {
+		
+		node * temp = head;
+		
+		for ( char c : input ) {
+			temp = add( temp, c );
+		}
+	}
+
+	// print out linked list 	
+	void show( linkedList * test ) {
+		
+		node * i = test->head;
+
+		while( i != NULL ) {
+			cout << i->val << endl;
+			i = i->next;
+		}
+
+	}
+
+	// check if linked list is palindrome
+	bool is_palindrome( linkedList * h ) {
+	
+		if ( h == NULL )
+			throw h;
+
+		node * p = h->head; // for iterating through list
+		node * mid_point = h->head;
+		int twice = 0; // will always be half the size of the list iteration
+		
+		stack<char> half_list;
+		
+		// find middle of linked list,
+		// add half of the list to a stack
+		while ( p != NULL ) {
```

Comment:
I know what you're doing here, but the naming of the variables is a bit confusing. What you're really doing is having two pointers, where one is moving 2x faster than the other. Rewrite it such that your 'slow' pointer pushes everything onto the stack, and your fast pointer just moves 2x until you go to the end